File: item.py

class Item:
	MAX_ITEM_ID = 63
	
	_itemID = 0
	
	__init__(name,x=0,y=0):
		self.itemID = Item._itemID
		if Item._itemID < Item.MAX_ITEM_ID:
			Item._itemID += 1
		else:
			Item._itemID = 0
		self.x = x
		self.y = y

File: web-app/src/app/_types/Item.ts
export class Item {
  constructor(itemData: Partial<Item>) {
    Object.assign(this, itemData);
  }

  name: string;
  description: string;
  rarity: string;
  hasCharges?: boolean;
  maxCharges?: number;
  currentCharges?: number;
  equippable?: boolean;
  isEquipped?: boolean;
  equipmentType?: string;
  equipmentTags?: string[];
}
